File: 1245-uri.c
#include <string.h>
#include <stdio.h>

int main(){
    char L;
    int sapatos[61];
    int N, M, pares;

    while(scanf("%d", &N) != EOF){
        pares = 0;
        memset(sapatos, 0, sizeof(sapatos));

        for(int i = 0; i < N; ++i){
            scanf("%d %c\n", &M, &L);

            if(L == 'D'){
                if(sapatos[M] < 0)  ++pares;
                ++sapatos[M];
            }else{
                if(sapatos[M] > 0)  ++pares;
                --sapatos[M];
            }
        }

        printf("%d\n", pares);
    }

    return 0;
}

File: 1245-uri.py
while True:
    try:
        N=int(input())
        if N==0:
            break
        pares = 0
        sapatos = [0 for c in range(61)]
        for i in range(N):
            M,L=input().split(' ')
            M=int(M)
            if(L == 'D'):
                if(sapatos[M] < 0):
                    pares += 1
                sapatos[M] += 1
            else:
                if(sapatos[M] > 0):
                    pares += 1
                sapatos[M] -= 1
        print(pares)
    except EOFError:
        break
        
